app/components/filters/select-multiple-filter.ts

```diff
@@ -37,19 +46,15 @@ export default class SelectMultipleFilterComponent extends FilterComponent {
   }
 
   @action
-  async selectChange(m: []) {
-    // Update UI
-    this.selected = m;
-
-    // Update queryParameters
-    let values = '';
-    for (let i = 0; i < m.length; i++) {
-      values += get(m[i], this.args.searchField) + seperator;
-    }
-    values = values.replace(regex, '');
+  async selectChange(selectedOptions: Option[]) {
+    this.selected = selectedOptions;
 
     this.updateQueryParams({
-      [this.args.queryParam]: values,
+      [this.args.queryParam]: serializeArray(
+        selectedOptions
+          .map((value) => value[this.args.searchField])
+          .filter(Boolean) as string[]
```

Comment:
`as string[]` is needed because TS doesn't narrow the type to thruthy only values while in practice it can't ever be false after the `.filter`.

Reply:
Yeah, I have to look for the syntax every time I use a filter ^^. I don't know why they can't handle it correctly after all these years